packages/mobile/src/provider/chat/ChatProvider.ts

```diff
@@ -274,18 +295,15 @@ export const uploadChatMessage = async (
       groupId: message.fileMetadata.appData.groupId,
       userDate: message.fileMetadata.appData.userDate,
       fileType: CHAT_MESSAGE_FILE_TYPE,
-      content: jsonContent,
+      content: content,
```

Comment:
Maybe we ought to do it a bit differently for chat. I was thinking to keep the parts that fit still in the content; And add a hard cut off on the message; That way we can then extend the message renderers to only fetch the full payload when expanded by a user;

And that way we don't need to fetch the payloads of every message when they're never rendered (older page, or conversation isn't opened soon anymore);

Reply:
I initially gave a similar thought about this to keep it that way and fetch the payload when we click on the expand button but I kept it similar to the way we do for posts at the moment. 

Maybe we refactor both the same way?

Reply:
Don't see a reason in refactoring those together. And for posts that will be much harder to do, as splitting up Rich Text articles is a lot more elaborate vs plain text.

So for posts I would prefer to wait till we got a final decision from the BE on the max size of a header.

Reply:
makes sense